src/RegisterDevicePage.tsx

```diff
@@ -295,8 +305,11 @@ export default (): React.ReactElement => {
         setShowCredentialSecretModal(true);
       })
       .catch((err) => {
+        const errorMessage = deviceRegistrationLimitReached
+          ? `The device registration limit is set to ${deviceRegistrationLimitFetcher.value} and there are too many registered devices already.`
+          : err.message;
```

Comment:
This "assumes" the error was due to the registration limit.
To do that, we have to fetch the total registered number of devices before hand.

I don't like this approach too much. If we know the limit was already reached, we should disable the registration of new devices (but allow re-registering an already existing one).
If we rely on the backend for validation, we should inspect the error message making sure the error is due to the registration limit and not other causes (like expired token or connectivity issues)

Reply:
Since the backend already returns an error code specific to the registration limit being reached, I update the PR to derive the error message from the error returned by the backend.
This way it's the backend's responsibility to validate whether the registration must fail due to the limit being reached.